### ISSUE TITLE
Knockout binding inside a component doesn't work

### ISSUE THREAD

**Issue by nkint** [davidstutz/bootstrap-multiselect#621]
Hi.

I'm trying to use this plugin inside a knockout component. The select is correctly showed but doesn't update the label on un button when the value is changed. Moreover, if with the developer console I make the original select visible, I notice that it does not update when I change the bootstrap multiselect (= data binding is not working).

I managed to make a jsfiddle with a complete example that shows the failing behaviour: http://jsfiddle.net/r608aw2m/

**Comment by golaat** [comment]
This is occurring because the value attribute on the option elements on your knockout binding for the select is not set.  The bootstrap-multiselect  depends on the value attribute.  
![select](https://cloud.githubusercontent.com/assets/6027610/10340960/8c44ec54-6ce0-11e5-9de8-3052f3bd51fe.png)

I have worked around this issue with knockout by using a bit of jquery to set the value attribute to the text of the dropdown list after it is bound; it is a hack, but it works :)
```javascript

var trackingAttributeFundLists = $("select.fund-list:not('.initialized')");
trackingAttributeFundLists.find("option").each(function (index) {
	this.setAttribute("value", this.text);
});

trackingAttributeFundLists.addClass("initialized");
```

**Comment by Aaron-P** [comment]
This appears to be due to how the button text selection works in multiselect versus how the knockout options bindings work.  In your example you are binding a set of objects to the options of the select and you specify the optionsText binding which correctly populates the text of the underlying select, but you don't specify the optionsValue binding.  Because that isn't specified knockout is using custom data handling to pass around the object as the value rather than a specific string/number since it doesn't handle full objects inside value="" attributes.

This seems to be breaking the relationship between the underlying observable and the multiselect component in some way, where the observable is not being updated properly when a value is selected.  The best workaround for this issue right now would be to specify the optionsValue binding so that the value attribute of the underlying select gets populated correctly.  If you still need the actual object you would then need to use a computed observable or something like that to re-acquire it from the selected value string/number.  The full solution would be to find the problem with how multiselect is handling object values in relation to knockout and fix that.

One additional issue with your example though, is that you are not using the knockout binding that multiselect defines, but rather manually initiating it on the onInit function.  Outside of the custom binding multiselect doesn't do any sort of sensing of knockout usage/other bindings/etc.  so even if the underlying issue were fixed your example component still wouldn't work.  I have updated your example to use the proper binding and workaround: https://jsfiddle.net/r608aw2m/2/
